Brace structure formatting

NoxMax opened the pull request:
There is no functional change to the code here.

I am working on this file part of a longer PR and this block was painful to look at. It's got a very long line and braces inside braces inside braces. I'm not sure what the best way to structure it. Here is my suggestion, but I am not sure if it's actually even better now. We got wrapped lines in this block that have 1 space indentation instead of 4 (which is kinda what's implied we should stick to according to the AC standard)

So if anything, I am making this PR to discuss and establish some precedence about how such long lines should be formatted in the future.

NoxMax commented on src/strategy/values/AttackersValue.cpp:
```diff
@@ -166,33 +166,53 @@ bool AttackersValue::IsPossibleTarget(Unit* attacker, Player* bot, float range)
+        !(attacker->IsPlayer() && !attacker->IsPvP() && !attacker->IsFFAPvP() &&
+            (!bot->duel || bot->duel->Opponent != attacker)) &&
+        (
+            !c ||
```
Or maybe keeping most of the original vertical spacing, but just making sure to always use 4 space indentation
```
        (!c ||
            (!c->IsInEvadeMode() &&
                ((!isMemberBotGroup && botAI->HasStrategy("attack tagged", BOT_STATE_NON_COMBAT)) ||
                    leaderHasThreat ||
                        (!c->hasLootRecipient() &&
                            (!c->GetVictim() ||
                                (c->GetVictim() &&
                                    ((!c->GetVictim()->IsPlayer() || bot->IsInSameGroupWith(c->GetVictim()->ToPlayer())) ||
                                        (botAI->GetMaster() && c->GetVictim() == botAI->GetMaster()))))) ||
                        c->isTappedBy(bot))));
```

hermensbas commented on src/strategy/values/AttackersValue.cpp:
```diff
@@ -166,33 +166,53 @@ bool AttackersValue::IsPossibleTarget(Unit* attacker, Player* bot, float range)
+                    c->isTappedBy(bot)
+                )
+            )
+        );
```
You can assign the conditions to a variable if you look for more readability instead of having the entire thing inline.

NoxMax replied:
I guess such example is approaching a limit that would require that. Maybe a standard to set is if a block requires some X number of indentations or more, then introducing extract variables would be best practice.

privatecore replied:
Formatting won't help here; the entire `AttackersValue::IsPossibleTarget` method needs to be rewritten. ;)

icemansparks replied:
I agree with that, the formatting makes it easier to look at, but the code is still very much a mess when it comes to understandable variable names 🙈

NoxMax commented:
Now edited with extract variables following Bash's suggestion. All in all, I think we should try to avoid going more than 8-10 indentations deep whenever possible, even 8 might be too much. And if something is returning more than 3 inline conditions, probably best to extract them.
Also I added comments because comments are always great, and removed returned deprecated code but kept the commented-out bools
